pkg/api/validation/validation.go

```diff
@@ -183,6 +183,13 @@ func ValidateEndpointsName(name string, prefix bool) (bool, string) {
 	return NameIsDNSSubdomain(name, prefix)
 }
 
+// ValidateEventName can be used to check whether the given event name is valid.
+// Prefix indicates this name will be used as part of generation, in which case
+// trailing dashes are allowed.
+func ValidateEventName(name string, prefix bool) (bool, string) {
+	return NameIsDNSSubdomain(name, prefix)
```

Comment:
Yeah, as of this change, events may not be created for 3rd party non-namespaced types. I don't think that's a desirable final state. I'm OK with it temporarily, but file an issue for tracking or this will be a regression in 1.2.